File: maximum-number-of-balls-in-a-box/maximum-number-of-balls-in-a-box.java
class Solution {
    public int countBalls(int lowLimit, int highLimit) {
        HashMap<Integer, Integer> boxToFreq = new HashMap<>();
        int maxSum = 0;
        for(int num = lowLimit; num <= highLimit; num++) {
            int sum = getSum(num);
            if(!boxToFreq.containsKey(sum)) {
                boxToFreq.put(sum, 1);
            } else {
                boxToFreq.put(sum, boxToFreq.get(sum) + 1);
            }
            if(maxSum == 0 || boxToFreq.get(sum) > boxToFreq.get(maxSum)) {
                    System.out.println(boxToFreq.get(sum));
                    maxSum = sum;   
            }
            
        }
        return (maxSum == 0) ? 0 : boxToFreq.get(maxSum);
    }
    
    private int getSum(int num) {
        int sum = 0;
        while(num != 0) {
            int digit = num % 10;
            sum += digit;
            num = (num - digit) / 10;
        }
        return sum;
    }
}

File: maximum-number-of-balls-in-a-box/maximum-number-of-balls-in-a-box.py
class Solution:
    def countBalls(self, lowLimit: int, highLimit: int) -> int:
        def getSum(num: int) -> int:
            sum = 0
            while num != 0:
                sum += num % 10
                num //= 10
            return sum
        
        boxes = [0] * 46
        maxId = 0;
        for num in range(lowLimit, highLimit + 1):
            boxId = getSum(num)
            boxes[boxId] += 1
            if boxes[boxId] > boxes[maxId]:
                maxId = boxId
        return boxes[maxId]
        